torchx/schedulers/ray/ray_driver.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import contextlib
+import importlib
+import json
+import logging
+import os
+import sys
+from typing import Dict, List, Optional
+
+import ray
+from ray.train.utils import get_address_and_port
+from ray.util.placement_group import PlacementGroup
+from torchx.schedulers.ray.ray_common import RayActor
+
+_logger: logging.Logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+@contextlib.contextmanager
+def redirect_argv(args: List[str]):  # pyre-ignore[3]
+    _argv = sys.argv[:]
+    sys.argv = args
+    yield
+    sys.argv = _argv
+
+
+@ray.remote
+class CommandActor:  # pragma: no cover
+    def __init__(self, command: str, env: Dict[str, str]) -> None:
+        self.args: List[str] = command.split(" ")
+        self.path: str = self.args[0]
+
+        for k, v in env.items():
+            os.environ[k] = v
+
+    def run_command(self) -> None:
+        spec: Optional[
+            importlib.machinery.ModuleSpec
+        ] = importlib.util.spec_from_file_location("__main__", self.path)
+        if spec:  # pragma: no cover
+            train = importlib.util.module_from_spec(spec)
+            with redirect_argv(self.args):
+                spec.loader.exec_module(train)  # pyre-ignore[16]
+
+
+def load_actor_json(filename: str) -> List[RayActor]:
+    with open(filename) as f:
+        actors: List[RayActor] = []
+        # Yes this is gross but it works
+        actor_dict = json.load(f)
+        actor_dict = json.loads(actor_dict)
```

Comment:
this works because your file is double-json encoded, e.g.:
```
# not real code
with open(filename) as f:
  f.write(json.write(json.write(my_obj)))
```

Why is it the case?

Reply:
```
    def serialize(
        actors: List[RayActor], dirpath: str, output_filename: str = "actors.json"
    ) -> None:
        # Takes List of Ray Actors dataclasses and turn them into JSON
        actors_json = json.dumps(actors, cls=_EnhancedJSONEncoder)


        with open(os.path.join(dirpath, output_filename), "w") as tmp:
                # Turn the list into a JSON
                json.dump(actors_json, tmp)

```